mlflow/pipelines/steps/train.py

```diff
@@ -699,17 +709,34 @@ def objective(hyperparameter_args):
         algo_type, algo_name = tuning_params["algorithm"].rsplit(".", 1)
         tuning_algo = getattr(importlib.import_module(algo_type, "hyperopt"), algo_name)
         max_trials = tuning_params["max_trials"]
-        hp_trials = Trials()
+        parallelism = tuning_params["parallelism"]
+
+        if parallelism > 1:
+            from hyperopt import SparkTrials
+            from mlflow.utils._spark_utils import _get_active_spark_session
+
+            spark_session = _get_active_spark_session()
```

Comment:
`_get_active_spark_session()` will return `None` if no Spark session is active. In this case, can we throw an exception indicating that an active Spark session must exist in order to perform tuning with `parallelism > 1`?